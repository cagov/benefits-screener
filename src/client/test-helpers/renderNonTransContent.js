import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
