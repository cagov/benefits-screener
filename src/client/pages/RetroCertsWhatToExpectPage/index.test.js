import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";
import AUTH_STRINGS from "../../../data/authStrings";

describe("<RetroCertsWhatToExpectPage />", () => {
  it("has user data", async () => {
    const wrapper = renderNonTransContent(Component, "RetroCertsWhatToExpectPage", {
      userData: {
        status: AUTH_STRINGS.statusCode.ok,
        lastName: "Lastname",
        weeksToCertify: [1, 2],
      }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
