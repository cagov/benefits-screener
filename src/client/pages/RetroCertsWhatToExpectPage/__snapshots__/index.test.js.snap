// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RetroCertsWhatToExpectPage /> has user data 1`] = `
<div
  className="what-to-expect"
  id="overflow-wrapper"
>
  <Header />
  <main>
    <div
      className="container p-4"
    >
      <h1>
        Retroactive Unemployment Certification
      </h1>
      <Alert
        closeLabel="Close alert"
        show={true}
        transition={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "appear": false,
              "in": false,
              "mountOnEnter": false,
              "timeout": 300,
              "unmountOnExit": false,
            },
            "displayName": "Fade",
            "render": [Function],
          }
        }
        variant="success"
      >
        <img
          alt="checkmark"
          className="checkmark"
          src="/images/check-circle-fill.svg"
        />
        We found your information!
      </Alert>
      <h2>
        What to expect
      </h2>
      <p>
        We'll ask you about your unemployment status, including the following:
      </p>
      <ul>
        <li>
          Were you too sick or injured to work?
        </li>
        <li>
          Was there any reason (other than sickness or injury) that you could not have accepted full-time work each work day?
        </li>
        <li>
          Did you look for work?
        </li>
        <li>
          Did you refuse any work?
        </li>
        <li>
          Did you begin attending any kind of school or training?
        </li>
        <li>
          Did you work or earn money, whether you were paid or not?
        </li>
      </ul>
      <p>
        You'll need to gather these details about your work, education, and other income for the time period from March 14 to May 9, 2020.
      </p>
      <h2>
        What if I already submitted my overpayments through AskEDD?
      </h2>
      <p>
        Please submit your eligibility information again.
      </p>
      <Button
        active={false}
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        disabled={false}
        to="/retroactive-certification/weeks-to-certify"
        type="button"
        variant="secondary"
      >
        Start certification
      </Button>
    </div>
  </main>
  <Footer />
</div>
`;
